remotes/docker/pusher_test.go

```diff
@@ -293,7 +293,7 @@ func Test_dockerPusher_push(t *testing.T) {
 		dp               dockerPusher
 		dockerBaseObject string
 		args             args
-		checkerFunc      func(writer pushWriter) bool
+		checkerFunc      func(writer *pushWriter) bool
```

Comment:
Why was this changed to a pointer?

Reply:
Executing `make check` without passing a pointer reports the following error:
```shell
➜  containerd git:(fix/push_concurrent_close_channel) make check
🇩 proto-fmt
🇩 check
GOGC=75 golangci-lint run
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
remotes/docker/pusher_test.go:309:29: copylocks: func passes lock by value: github.com/containerd/containerd/remotes/docker.pushWriter contains sync.Once contains sync.Mutex (govet)
			checkerFunc: func(writer pushWriter) bool {
			                         ^
remotes/docker/pusher_test.go:343:29: copylocks: func passes lock by value: github.com/containerd/containerd/remotes/docker.pushWriter contains sync.Once contains sync.Mutex (govet)
			checkerFunc: func(writer pushWriter) bool {
			                         ^
remotes/docker/pusher_test.go:382:36: copylocks: call of test.checkerFunc copies lock value: github.com/containerd/containerd/remotes/docker.pushWriter contains sync.Once contains sync.Mutex (govet)
			assert.True(t, test.checkerFunc(*pw))
			                                ^
make: *** [check] Error 1
```

Reply:
Wondering why this does not show up in the CI. A quick look shows that this is [executed](https://github.com/containerd/containerd/blob/main/script/setup/install-protobuf#L57) only when the GOARCH is not amd64, arm64 or ppc64le. This script is being called as part of Protobuf installation in [CI](https://github.com/containerd/containerd/blob/main/.github/workflows/ci.yml#L99)